src/l500/l500-color.cpp

```diff
@@ -493,10 +493,44 @@ namespace librealsense
         return tags;
     }
 
-    std::vector<uint8_t> l500_color::get_raw_intrinsics_table() const
+    ivcam2::intrinsic_rgb l500_color::get_intrinsics_table() const
     {
-        AC_LOG( DEBUG, "RGB_INTRINSIC_GET" );
-        return _hw_monitor->send(command{ RGB_INTRINSIC_GET });
+        // Get RAW data from firmware
+        std::vector< uint8_t > response_vec = _hw_monitor->send(command{ RGB_INTRINSIC_GET });
+
+        if (response_vec.size() == 0)
+            throw invalid_value_exception("Calibration data invalid,buffer size is zero");
+
+        auto resolutions_rgb_table_ptr = reinterpret_cast<const ivcam2::intrinsic_rgb *>(response_vec.data());
+
+        // Get full maximum size of the resolution array and deduct the unused resolutions size from it
```

Comment:
I suggest you comment what the "agreement" is between FW and us: that they guarantee only as many bytes as required for the given number of resolutions, and that **they may send more**.

Reply:
Added